src/gui/optionsdlg.cpp

```diff
@@ -64,12 +64,116 @@
 #include "addnewtorrentdialog.h"
 #include "advancedsettings.h"
 #include "rss/automatedrssdownloader.h"
+#include "theme/themeinfo.h"
+#include "theme/themeprovider.h"
 #include "banlistoptions.h"
 #include "guiiconprovider.h"
 #include "scanfoldersdelegate.h"
 
+#include <algorithm>
+#include <vector>
+
 #include "ui_optionsdlg.h"
 
+namespace
+{
+    /**
+     * @brief Implements list mode for themes of the give kind
+     *
+     * The class gets map of the themes from ThemeProvider, converts it into array sorting
+     * by theme localized name.
+     */
+    class ThemeInfoListModel: public QAbstractListModel {
+    public:
+        ThemeInfoListModel(Theme::Kind kind, QObject *parent);
+
+        int rowCount(const QModelIndex &parent) const override;
+        QVariant data(const QModelIndex &index, int role) const override;
+
+        struct Entry {
+            Theme::ThemeInfo info;
+            QString filePath;
+        };
+
+        const Entry &entry(int index) const;
+
+        int indexForTheme(const Theme::ThemeInfo &info);
+    private:
+        std::vector<Entry> m_entries;
+    };
+
+    ThemeInfoListModel::ThemeInfoListModel(Theme::Kind kind, QObject *parent)
+        : QAbstractListModel(parent)
+    {
+        std::map<Theme::ThemeInfo, QString> themes = Theme::ThemeProvider::instance().availableThemes(kind);
+        for (const auto& pair: themes) {
+            m_entries.push_back({pair.first, pair.second});
+        }
+        std::sort(m_entries.begin(), m_entries.end(),
+                  [](const Entry &left, const Entry &right) {
+                      return left.info.localizedName() < right.info.localizedName();
+                  });
+    }
+
+    int ThemeInfoListModel::rowCount(const QModelIndex &/*parent*/) const
+    {
+        return static_cast<int>(m_entries.size());
+    }
+
+    QVariant ThemeInfoListModel::data(const QModelIndex &index, int role) const
+    {
+        if (index.row() >= static_cast<int>(m_entries.size()) || index.column() > 0) return {};
+
+        switch (role) {
+            case Qt::DisplayRole:
+                return entry(index.row()).info.localizedName();
+            case Qt::ToolTipRole:
+                return entry(index.row()).info.localizedDescription();
+        }
+
+        return {};
+    }
+
+    const ThemeInfoListModel::Entry &ThemeInfoListModel::entry(int index) const
+    {
+        return m_entries.at(static_cast<std::size_t>(index));
```

Comment:
Because `QAbstractItemModel` interface uses `int` and replacing parameter type we get casts in `data()` then.